daliuge-engine/dlg/apps/pyfunc.py

```diff
@@ -712,18 +712,25 @@ def initialize_with_func_code(self):
         """
         This function takes over if code is passed in through an argument.
         """
+        b64reg = (
```

Comment:
I think this function needs further documentation to properly describe what it is achieving, and why we might expect serialized or unserialized functions. 

We should probably add a unit test for this too to test all the options and make it clear what the expectations are.